src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElastiSearchAnalysisGraph.java

```diff
@@ -88,5 +94,32 @@ public void construct() {
     highHeapUsageClusterRca.addTag(TAG_LOCUS, LOCUS_MASTER_NODE);
     highHeapUsageClusterRca.addAllUpstreams(Collections.singletonList(hotJVMNodeRca));
     highHeapUsageClusterRca.addTag(TAG_AGGREGATE_UPSTREAM, LOCUS_DATA_NODE);
+
+    createShardResourceUsageGraph();
+  }
+
+  private void createShardResourceUsageGraph() {
+    Metric cpuUsage = new CPU_Utilization(5);
+    Metric ioTotThroughput = new IO_TotThroughput(5);
+    Metric ioTotSyscallRate = new IO_TotalSyscallRate(5);
+
+
+    cpuUsage.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    ioTotThroughput.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    ioTotSyscallRate.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    addLeaf(cpuUsage);
+    addLeaf(ioTotThroughput);
+    addLeaf(ioTotSyscallRate);
+
+    // High CPU Usage RCA
+    HighCPUShardRca highCPUShardRca = new HighCPUShardRca(5, 12, cpuUsage, ioTotThroughput, ioTotSyscallRate);
+    highCPUShardRca.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    highCPUShardRca.addAllUpstreams(Arrays.asList(cpuUsage, ioTotThroughput, ioTotSyscallRate));
+
+    // Hot Shard RCA which consumes the HighCPUShardRca
+    HotShardClusterRca hotShardClusterRca = new HotShardClusterRca(12, highCPUShardRca);
+    hotShardClusterRca.addTag(TAG_LOCUS, LOCUS_MASTER_NODE);
+    hotShardClusterRca.addAllUpstreams(Collections.singletonList(highCPUShardRca));
+    hotShardClusterRca.addTag(RcaTagConstants.TAG_AGGREGATE_UPSTREAM, LOCUS_DATA_NODE);
```

Comment:
do we still need to set TAG_AGGREGATE_UPSTREAM on master node ? I assume HighCPUShardRca will only run on data node right ?

Reply:
As discussed, we will be needing the `TAG_AGGREGATE_UPSTREAM` here.